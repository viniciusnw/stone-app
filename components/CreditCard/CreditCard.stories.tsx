import React from "react";
import { CreditCard, StorybookWrapper } from "@Components";

export type Props = {
  name: string;
  number: string;
  brand: "visa" | "mastercard" | "santander";
};

const card: Props = {
  name: "Bruce Wayne",
  number: "**** **** **** 1234",
  brand: "visa",
};

const card1: Props = {
  name: "Dick Grayson",
  number: "**** **** **** 1234",
  brand: "mastercard",
};

export const Default = {};

export default {
  title: "Credit Card",
  component: (props: any) => (
    <StorybookWrapper>
      <CreditCard {...card} />
      <CreditCard {...card1} />
    </StorybookWrapper>
  ),
};
