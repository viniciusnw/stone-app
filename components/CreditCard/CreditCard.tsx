import React from 'react';
import { View, Text, Image } from 'react-native';
import styled from 'styled-components/native';
import { icons } from '@Assets';

const StyledContainer = styled(View)`
  width: 100%;
  flex-direction: row;
  align-items: center;
  border: 1px solid ${({ theme }) => theme.color.border};
  padding: 0 ${({ theme }) => theme.spacing.regular};
  margin-bottom: ${({ theme }) => theme.spacing.regular};
`;
const StyledWrapper = styled(View)`
  flex-direction: row;
  justify-content: flex-start;
  flex: 1;
`;

const StyledInfo = styled(View)`
  margin-left: ${({ theme }) => theme.spacing.regular};
  padding: ${({ theme }) => theme.spacing.regular};
  border: 1px solid transparent;
  border-left-color: ${({ theme }) => theme.color.border};
`;

const StyledLogo = styled(View)`
  justify-content: center;
`;

const StyledName = styled(Text)`
  font-size: 18px;
  font-weight: 700;
  margin-bottom: 5px;
  color: ${({ theme }) => theme.color.gray};
`;

const StyledNumber = styled(Text)`
  font-size: 13px;
  font-weight: 700;
  color: ${({ theme }) => theme.color.gray};
`;

type Props = {
  name: string,
  number: string,
  brand: 'visa' | 'mastercard' | 'santander',
}

const CreditCard:React.FC<Props> = ({ name, number, brand }) => {
  return (
    <StyledContainer>
      <StyledWrapper>
        <StyledLogo>
          <Image style={{ width: 52, height: 52 }} source={icons[brand]} />
        </StyledLogo>

        <StyledInfo>
          <StyledName>{name}</StyledName>
          <StyledNumber>{number}</StyledNumber>
        </StyledInfo>
      </StyledWrapper>
    </StyledContainer>
  );
};

export default CreditCard;
