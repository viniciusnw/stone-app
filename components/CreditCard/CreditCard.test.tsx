import React from "react";
import renderer from "react-test-renderer";
import { darkTheme, lightTheme } from "@Theme";
import { ThemeProvider } from "styled-components/native";
import "jest-styled-components";
import CreditCard from "./CreditCard";

jest.useFakeTimers();

type Props = {
  name: string;
  number: string;
  brand: "visa" | "mastercard" | "santander";
};

const card: Props = {
  name: "Bruce Wayne",
  number: "**** **** **** 1234",
  brand: "visa",
};

const card1: Props = {
  name: "Dick Grayson",
  number: "**** **** **** 1234",
  brand: "mastercard",
};

test("CreditCard", () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={darkTheme}>
        <CreditCard {...card} />
        <CreditCard {...card1} />
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
