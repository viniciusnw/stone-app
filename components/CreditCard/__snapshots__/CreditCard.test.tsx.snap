// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreditCard 1`] = `
[
  <View
    style={
      {
        "alignItems": "center",
        "borderColor": "#ddd",
        "borderStyle": "solid",
        "borderWidth": 1,
        "flexDirection": "row",
        "marginBottom": 15,
        "paddingBottom": 0,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 0,
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "flexBasis": 0,
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "flex-start",
        }
      }
    >
      <View
        style={
          {
            "justifyContent": "center",
          }
        }
      >
        <Image
          source={1}
          style={
            {
              "height": 52,
              "width": 52,
            }
          }
        />
      </View>
      <View
        style={
          {
            "borderColor": "transparent",
            "borderLeftColor": "#ddd",
            "borderStyle": "solid",
            "borderWidth": 1,
            "marginLeft": 15,
            "paddingBottom": 15,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 15,
          }
        }
      >
        <Text
          style={
            {
              "color": "#666666",
              "fontSize": 18,
              "fontWeight": "700",
              "marginBottom": 5,
            }
          }
        >
          Bruce Wayne
        </Text>
        <Text
          style={
            {
              "color": "#666666",
              "fontSize": 13,
              "fontWeight": "700",
            }
          }
        >
          **** **** **** 1234
        </Text>
      </View>
    </View>
  </View>,
  <View
    style={
      {
        "alignItems": "center",
        "borderColor": "#ddd",
        "borderStyle": "solid",
        "borderWidth": 1,
        "flexDirection": "row",
        "marginBottom": 15,
        "paddingBottom": 0,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 0,
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "flexBasis": 0,
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "flex-start",
        }
      }
    >
      <View
        style={
          {
            "justifyContent": "center",
          }
        }
      >
        <Image
          source={1}
          style={
            {
              "height": 52,
              "width": 52,
            }
          }
        />
      </View>
      <View
        style={
          {
            "borderColor": "transparent",
            "borderLeftColor": "#ddd",
            "borderStyle": "solid",
            "borderWidth": 1,
            "marginLeft": 15,
            "paddingBottom": 15,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 15,
          }
        }
      >
        <Text
          style={
            {
              "color": "#666666",
              "fontSize": 18,
              "fontWeight": "700",
              "marginBottom": 5,
            }
          }
        >
          Dick Grayson
        </Text>
        <Text
          style={
            {
              "color": "#666666",
              "fontSize": 13,
              "fontWeight": "700",
            }
          }
        >
          **** **** **** 1234
        </Text>
      </View>
    </View>
  </View>,
]
`;
